frontend/src/resources/managedclusteraction.tsx

```diff
@@ -0,0 +1,169 @@
+    resourceName: string,
+    resourceNamespace: string,
+    resourceBody?: any
+) => {
+    const actionName = crypto
+        .createHash('sha1')
+        .update(`${actionType}-${resourceName}-${resourceKind}`)
+        .digest('hex')
+        .substr(0, 63)
+    const { apiGroup, version } = getGroupFromApiVersion(resourceApiVersion)
+    return createResource<ManagedClusterAction>({
+        apiVersion: ManagedClusterActionApiVersion,
+        kind: ManagedClusterActionKind,
+        metadata: {
+            name: actionName,
+            namespace: clusterName,
+        },
+        spec: {
+            cluster: {
+                name: clusterName,
+            },
+            type: 'Action',
+            scope: {
+                resourceType: apiGroup
+                    ? `${resourceKind.toLowerCase()}.${version}.${apiGroup}`
+                    : `${resourceKind.toLowerCase()}`,
+                namespace: resourceNamespace,
+            },
+            actionType: actionType,
+            kube: {
+                resource: apiGroup
+                    ? `${resourceKind.toLowerCase()}.${version}.${apiGroup}`
+                    : `${resourceKind.toLowerCase()}`,
+                name: resourceName,
+                namespace: resourceNamespace,
+                template: resourceBody,
+            },
+        },
+    })
+        .promise.then(async () => {
+            return pollManagedClusterAction(actionName, clusterName)
+        })
+        .catch((err) => {
+            console.error(err)
+            return err
+        })
+}
+
+export async function pollManagedClusterAction(actionName: string, clusterName: string): Promise<ManagedClusterAction> {
+    let retries = process.env.NODE_ENV === 'test' ? 0 : 20
+    const poll = async (resolve: any, reject: any) => {
+        getManagedClusterAction({ namespace: clusterName, name: actionName })
+            .promise.then((actionResponse) => {
+                const isComplete = _.get(actionResponse, 'status.conditions[0].type', '')
+                const isActionDone = _.get(actionResponse, 'status.conditions[0].reason', '')
+                const actionMessage = _.get(actionResponse, 'status.conditions[0].message', '')
+                if (isComplete === 'Completed' && isActionDone === 'ActionDone') {
+                    resolve({
+                        complete: isComplete,
+                        actionDone: isActionDone,
+                        message: actionMessage,
+                        result: actionResponse.status?.result,
+                    })
+                } else if (isComplete === 'Completed' && isActionDone !== 'ActionDone') {
+                    reject({ message: actionMessage })
+                }
+                deleteManagedClusterAction({ name: actionName, namespace: clusterName })
+            })
+            .catch((err) => {
+                if (retries-- > 0) {
+                    setTimeout(poll, 500, resolve, reject)
```

Comment:
Should we make the poll interval configurable? In console-api, looks like the default was 100, not 500.

Reply:
I can update it to 100. In looking at the console-api deployment we are never overriding the default value.